Maker-Agents-Suit-Module/OSMAgent/src/big.queued.price.deviation.spec.ts

```diff
@@ -0,0 +1,296 @@
+      })
+      .addEventLog(
+        "LogValue(bytes32)",
+        CONTRACT_ADDRESSES[0],
+        [],
+        web3.eth.abi.encodeParameter("uint128", 100)
+      );
+
+    const findings: Finding[] = await handleTransaction(txEvent);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should returns empty findings if the Peek function wasn't called from the correct contract", async () => {
+    handleTransaction = provideBigQueuedPriceDeviationHandler(
+      CONTRACT_ADDRESSES
+    );
+
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .addTrace({
+        from: CONTRACT_ADDRESSES[1],
+        input: PEEK_FUNCTION_SELECTOR,
+        output: web3.eth.abi.encodeParameters(["uint128", "bool"], [108, true]),
+      })
+      .addEventLog(
+        "LogValue(bytes32)",
+        CONTRACT_ADDRESSES[0],
+        [],
+        web3.eth.abi.encodeParameter("uint128", 100)
+      );
+
+    const findings: Finding[] = await handleTransaction(txEvent);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should returns empty findings if Peek function wasn't called", async () => {
+    handleTransaction = provideBigQueuedPriceDeviationHandler(
+      CONTRACT_ADDRESSES
+    );
+
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .addTrace({
+        from: CONTRACT_ADDRESSES[0],
+        input: "0x11111111",
+        output: web3.eth.abi.encodeParameters(["uint128", "bool"], [108, true]),
+      })
+      .addEventLog(
+        "LogValue(bytes32)",
+        CONTRACT_ADDRESSES[0],
+        [],
+        web3.eth.abi.encodeParameter("uint128", 100)
+      );
+
+    const findings: Finding[] = await handleTransaction(txEvent);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should returns empty findings if the Peek call wasn't successful", async () => {
+    handleTransaction = provideBigQueuedPriceDeviationHandler(
+      CONTRACT_ADDRESSES
+    );
+
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .addTrace({
+        from: CONTRACT_ADDRESSES[0],
+        input: PEEK_FUNCTION_SELECTOR,
+        output: web3.eth.abi.encodeParameters(
+          ["uint128", "bool"],
+          [108, false]
+        ),
+      })
+      .addEventLog(
+        "LogValue(bytes32)",
+        CONTRACT_ADDRESSES[0],
+        [],
+        web3.eth.abi.encodeParameter("uint128", 100)
+      );
+
+    const findings: Finding[] = await handleTransaction(txEvent);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should returns empty findings if the correct event wasn't logged", async () => {
+    handleTransaction = provideBigQueuedPriceDeviationHandler(
+      CONTRACT_ADDRESSES
+    );
+
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .addTrace({
+        from: CONTRACT_ADDRESSES[0],
+        input: PEEK_FUNCTION_SELECTOR,
+        output: web3.eth.abi.encodeParameters(["uint128", "bool"], [108, true]),
```

Comment:
Same here as above, I think that the problem is that you wrote tests for different `lenght` in the results of next and current, being consistent with the current implementation